### ISSUE TITLE
[Doc] Parametrize matrix order appers inverted in doc

### ISSUE THREAD

**Issue by GeorgeFischhof** [pytest-dev/pytest#3001]
[Doc] Parametrize matrix order appers inverted in doc

Hi there,

the page on the following url (and maybe other parts are affected too):
https://docs.pytest.org/en/latest/parametrize.html
says :

```
import pytest
@pytest.mark.parametrize("x", [0, 1])
@pytest.mark.parametrize("y", [2, 3])
def test_foo(x, y):
    pass
```

```
This will run the test with the arguments set to x=0/y=2, x=0/y=3, x=1/y=2 and x=1/y=3
```

But the inner and outer cycles are  in reverse order; the actual and expected result is:
```
test_foo[2-0]
test_foo[2-1]
test_foo[3-0]
test_foo[3-1]
```

Used:
Python 3.6.1
latest pytest
windows 7

BR,
George

**Comment by RonnyPfannschmidt** [comment]
good find, due to how python decorators work, declaration order and application order are reversed and we should make a note of that

**Comment by stsatlantis** [comment]
Should be this one resolved?

**Comment by nicoddemus** [comment]
Yes, thanks for the ping! Fixed by #3043
